fc/mef.py

```diff
@@ -217,265 +52,152 @@ def clustering_gmm(data, n_clusters = 8, initialization = 'distance_sub',
+    means = []
+    covars = []
+
+    # Get distance and sort based on it
+    dist = np.sum(data**2., axis=1)
+    sorted_i = np.argsort(dist)
+
+    # Expected number of elements per cluster
+    n_per_cluster = data.shape[0]/float(n_clusters)
+
+    # Get means and covariances per cluster
+    # We will just use a fraction of ``1 - 2*discard_frac`` of the data.
+    # Data at the edges that actually corresponds to another cluster can
+    # really mess up the final result.
+    discard_frac = 0.25
+    for i in range(n_clusters):
+        il = int((i + discard_frac)*n_per_cluster)
+        ih = int((i + 1 - discard_frac)*n_per_cluster)
+        sorted_i_i = sorted_i[il:ih]
+        data_i = data[sorted_i_i]
+        means.append(np.mean(data_i, axis=0))
         if data.shape[1] == 1:
-            covars = [np.cov(di.T).reshape(1,1) for di in data_clustered]
+            covars.append(np.cov(data_i.T).reshape(1,1))
         else:
-            covars = [np.cov(di.T) for di in data_clustered]
-
-        # Initialize GMM object
-        gmm = GMM(n_components = n_clusters, tol = tol, min_covar = min_covar,
-            covariance_type = 'full', params = 'mc', init_params = '')
-        gmm.weight_ = weights
-        gmm.means_ = means
-        gmm.covars_ = covars
-
-    elif initialization == 'distance_sub':
-        # Initialize parameters for GMM
-        weights = np.tile(1.0 / n_clusters,
-                                        n_clusters)
-        means = []
-        covars = []
-
-        # Get distance and sort based on it
-        dist = np.sum(data**2., axis = 1)
-        sorted_i = np.argsort(dist)
-
-        # Expected number of elements per cluster
-        n_per_cluster = data.shape[0]/float(n_clusters)
-
-        # Get the means and covariances per cluster
-        # We will just use a fraction of 1-2*discard_frac of the data.
-        # Data at the edges that actually corresponds to another cluster can
-        # really mess up the final result.
-        discard_frac = 0.25
-        for i in range(n_clusters):
-            il = int((i+discard_frac)*n_per_cluster)
-            ih = int((i+1-discard_frac)*n_per_cluster)
-            sorted_i_i = sorted_i[il:ih]
-            data_i = data[sorted_i_i]
-            means.append(np.mean(data_i, axis = 0))
-            if data.shape[1] == 1:
-                covars.append(np.cov(data_i.T).reshape(1,1))
-            else:
-                covars.append(np.cov(data_i.T))
-        means = np.array(means)
-
-        # Initialize GMM object
-        gmm = GMM(n_components = n_clusters, tol = tol, min_covar = min_covar,
-            covariance_type = 'full', params = 'mc', init_params = '')
-        gmm.weight_ = weights
-        gmm.means_ = means
-        gmm.covars_ = covars
-
-    else:
-        raise ValueError('Initialization method {} not implemented.'\
-            .format(initialization))
+            covars.append(np.cov(data_i.T))
+    means = np.array(means)
```

Comment:
this was a little unclear to read through. this is minor though. some possible alternative variable names:
* `sorted_i` -> `sorted_dist_idx`? `dist_idx_sorted`?
* `i` -> `subpop_idx`? `cluster_idx`?
* `il` -> `subpop_low_event_idx`, `ih` -> `subpop_high_event_idx`? or `cluster` in place of `subpop`? or `sample` in place of `event`?
* `sorted_i_i` -> `subpop_frac_idx`? `cluster_frac_idx`?
* `data_i` -> `data_subpop`? `data_cluster`? `data_cluster_frac`?

`discard_frac` also seems kinda like a misnomer? You're discarding 50%, not 25%. Could just do `discard_frac/2` everywhere and have `discard_frac=0.5`. This is really not very important, though.

Not really sure I like any of these suggestions. It was just a little confusing at first (`sorted_i_i` is pretty terrible, though, haha).